### ISSUE TITLE
Rancher-Agent crashes devicemapper with docker daemon

### ISSUE THREAD

**Issue by thirunaresh** [rancher/rancher#5956]
**Rancher Version:** 1.1.0

**Docker Version:** 1.10.3

OS and where are the hosts located? (cloud, bare metal, etc): RHEl/7.2/vmware

Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB) single node with external DB

**Environment Type:** Cattle

**Steps to Reproduce:**
Docker environment using devicemapper as storage driver with loopback device.

**Docker info:** 

Server Version: 1.10.3
Storage Driver: devicemapper
 Pool Name: docker-253:0-1180431-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop2
 Metadata file: /dev/loop3
 Data Space Used: 581 MB
 Data Space Total: 107.4 GB
 Data Space Available: 19.81 GB
 Metadata Space Used: 1.434 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.146 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2015-10-14)
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: null host bridge
Kernel Version: 3.10.0-327.4.4.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.2 (Maipo)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 31.26 GiB

**Results:**
**Containers went to unresponsive state.**
**Stopping or deleting container will end container in "dead" state.**
**Restarting docker daemon failing with below message.**
Sep 12 03:58:27 sv1lxdcim04 docker[7434]: time="2016-09-12T03:58:27.172380678-07:00" level=warning msg="devmapper: Usage of loopback devices is strongly discouraged for production use. Please use `--storage-opt dm.thinpooldev` or use `man docker` to refer to dm.thinpooldev section."
Sep 12 03:58:27 sv1lxdcim04 docker[7434]: time="2016-09-12T03:58:27.226281019-07:00" level=error msg="[graphdriver] prior storage driver \"devicemapper\" failed: devmapper: Base Device UUID and Filesystem verification failed.devmapper: Failed to find uuid for device /dev/mapper/docker-253:2-5767170-base:exit status 2"
Sep 12 03:58:27 sv1lxdcim04 docker[7434]: time="2016-09-12T03:58:27.226350604-07:00" level=fatal msg="Error starting daemon: error initializing graphdriver: devmapper: Base Device UUID and Filesystem verification failed.devmapper: Failed to find uuid for device /dev/mapper/docker-253:2-5767170-base:exit status 2"
Sep 12 03:58:27 sv1lxdcim04 systemd[1]: docker.service: main process exited, code=exited, status=1/FAILURE
Sep 12 03:58:27 sv1lxdcim04 systemd[1]: Failed to start Docker Application Container Engine.

cadvisor process still showing as zombie process, even after docker daemon is down.

Docker daemon running back only after cleaning devicemapper directory. which clears all the image and container data

**Expected:**

**Comment by thirunaresh** [comment]
This issue looks more related to cadvisor, as mentioned in these references.

Ref: https://github.com/docker/docker/issues/9665#issuecomment-137422559
       https://github.com/google/cadvisor/issues/771

**Comment by aemneina** [comment]
definitely a known issue... also avoid loopback driver for devicemapper.
